classes/oai/ojs/OAIDAO.inc.php

```diff
@@ -252,72 +242,90 @@ public function _getRecordsRecordSet($setIds, $from, $until, $set, $submissionId
-				' . ($until ? ' AND GREATEST(a.last_modified, i.last_modified) <= ' . $this->datetimeToDB($until) : '') . '
-				' . ($submissionId ? ' AND a.submission_id = ?' : '') . '
-			UNION
-			SELECT	dot.date_deleted AS last_modified,
-				dot.data_object_id AS submission_id,
-				' . (isset($journalId) ? 'tsoj.assoc_id' : 'NULL') . ' AS assoc_id,' . '
-				' . (isset($sectionId) ? 'tsos.assoc_id' : 'NULL') . ' AS section_id,
-				NULL AS issue_id,
-				dot.tombstone_id,
-				dot.set_spec,
-				dot.oai_identifier
-			FROM	data_object_tombstones dot' . '
-				' . (isset($journalId) ? 'JOIN data_object_tombstone_oai_set_objects tsoj ON (tsoj.tombstone_id = dot.tombstone_id AND tsoj.assoc_type = ' . ASSOC_TYPE_JOURNAL . ' AND tsoj.assoc_id = ?)' : '') . '
-				' . (isset($sectionId) ? 'JOIN data_object_tombstone_oai_set_objects tsos ON (tsos.tombstone_id = dot.tombstone_id AND tsos.assoc_type = ' . ASSOC_TYPE_SECTION . ' AND tsos.assoc_id = ?)' : '') . '
-			WHERE	1=1
-				' . (isset($set) ? ' AND (dot.set_spec = ? OR dot.set_spec LIKE ?)' : '') . '
-				' . ($from ? ' AND dot.date_deleted >= ' . $this->datetimeToDB($from) : '') . '
-				' . ($until ? ' AND dot.date_deleted <= ' . $this->datetimeToDB($until) : '') . '
-				' . ($submissionId ? ' AND dot.data_object_id = ?' : '') . '
-			ORDER BY ' . $orderBy,
-            $params
-        );
+        return DB::table('submissions AS a')
+            ->select([
+                DB::raw('GREATEST(a.last_modified, i.last_modified) AS last_modified'),
+                'a.submission_id AS submission_id',
+                'i.issue_id',
+                DB::raw('NULL AS tombstone_id'),
+                DB::raw('NULL AS set_spec'),
+                DB::raw('NULL AS oai_identifier'),
+                'j.journal_id AS journal_id',
+                's.section_id AS section_id',
+            ])
+            ->join('publications AS p', 'a.current_publication_id', '=', 'p.publication_id')
+            ->join('publication_settings AS psissue', function($join) {
+                $join->on('psissue.publication_id', '=', 'p.publication_id');
+                $join->where('psissue.setting_name', '=', DB::raw('\'issueId\''));
+                $join->where('psissue.locale', '=', DB::raw('\'\''));
+            })
+            ->join('issues AS i', DB::raw('CAST(i.issue_id AS CHAR(20))'), '=', 'psissue.setting_value')
+            ->join('sections AS s', 's.section_id', '=', 'p.section_id')
+            ->join('journals AS j', 'j.journal_id', '=', 'a.context_id')
+            ->where('i.published', '=', 1)
+            ->where('j.enabled', '=', 1)
+            ->where('a.status', '=', PKPSubmission::STATUS_PUBLISHED)
+            ->when($excludeJournals, function($query, $excludeJournals) {
+                return $query->whereNotIn('j.journal_id', $excludeJournals);
+            })
+            ->when(isset($journalId), function($query) use ($journalId) {
```

Comment:
Why sometimes with and sometimes without 'use'?

Reply:
This keeps the new code consistent with the prior tests for `if ($myVariable) ...` vs. `if (isset($myVariable)) ...`
I'm not sure if the difference was originally there for a reason and didn't want to introduce regressions.